rectools/models/nn/transformer_base.py

```diff
@@ -855,3 +890,73 @@ def _get_config(self) -> TransformerModelConfig_T:
         params = {attr: getattr(self, attr) for attr in attrs if attr != "cls"}
         params["cls"] = self.__class__
         return self.config_class(**params)
+
+    def __getstate__(self) -> object:
+        if self.is_fitted:
+            with NamedTemporaryFile() as f:
+                self.fit_trainer.save_checkpoint(f.name)
+                state = torch.load(f.name, weights_only=False)
+                state["is_fitted"] = True
+            return state
+        state = {"model_config": self.get_config(simple_types=True), "is_fitted": False, "trainer": self._trainer}
```

Comment:
Why `simple_types=True`? This doesn't allow you to save random state, e.g. And I can't see any benefits

Reply:
fixed